src/javascript/class-scanner.ts

```diff
@@ -594,70 +668,72 @@ export function extractPropertiesFromConstructor(
     return properties;
   }
   for (const method of astNode.body.body) {
-    if (!babel.isClassMethod(method) || method.kind !== 'constructor') {
+    if (!babel.isClassMethod(method)) {
       continue;
     }
-    const constructor = method;
-    for (const statement of constructor.body.body) {
-      if (!babel.isExpressionStatement(statement)) {
-        continue;
-      }
-      let name;
-      let astNode;
-      let defaultValue;
-      if (babel.isAssignmentExpression(statement.expression)) {
-        // statements like:
-        // /** @public The foo. */
-        // this.foo = baz;
-        name = getPropertyNameOnThisExpression(statement.expression.left);
-        astNode = statement.expression.left;
-        defaultValue = generate(statement.expression.right).code;
-      } else if (babel.isMemberExpression(statement.expression)) {
-        // statements like:
-        // /** @public The foo. */
-        // this.foo;
-        name = getPropertyNameOnThisExpression(statement.expression);
-        astNode = statement;
-      } else {
-        continue;
+
+    if (method.kind === 'constructor') {
+      const constructor = method;
+      for (const statement of constructor.body.body) {
+        if (!babel.isExpressionStatement(statement)) {
+          continue;
+        }
+        const prop = extractPropertyFromExpressionStatement(statement, document);
+        if (!prop) {
+          continue;
+        }
+        properties.set(prop.name, prop);
       }
-      if (name === undefined) {
+    } else if (method.kind === 'get' || method.kind === 'set') {
+      const prop = extractPropertyFromMethod(method, document);
```

Comment:
Likewise, this isn't extracting a property from a method, it's extracting a property from a getter or setter. Otherwise we should emit a `Method` not a `Property`